SpatialGDK/Source/SpatialGDKServices/Private/SSpatialOutputLog.cpp

```diff
@@ -254,21 +255,58 @@ void SSpatialOutputLog::StartPollTimer(const FString& LogFilePath)
 	});
 }
 
+void SSpatialOutputLog::FormatAndPrintRawErrorLine(const FString& LogLine)
+{
+	const FRegexPattern ErrorPattern = FRegexPattern(TEXT("level=(.*) msg=(.*) code=(.*) code_string=(.*) error=(.*) stack=(.*)"));
+	FRegexMatcher ErrorMatcher(ErrorPattern, LogLine);
+
+	if (!ErrorMatcher.FindNext())
+	{
+		UE_LOG(LogSpatialOutputLog, Error, TEXT("Failed to parse log line: %s"), *LogLine);
+		return;
+	}
+
+	FString ErrorLevelText = ErrorMatcher.GetCaptureGroup(1);
+	FString Message = ErrorMatcher.GetCaptureGroup(2);
+	FString ErrorCode = ErrorMatcher.GetCaptureGroup(3);
+	FString ErrorCodeString = ErrorMatcher.GetCaptureGroup(4);
+	FString ErrorMessage = ErrorMatcher.GetCaptureGroup(5);
+	FString Stack = ErrorMatcher.GetCaptureGroup(6);
+
+	// The stack message comes with double escaped characters.
+	Stack = Stack.ReplaceEscapedCharWithChar();
+
+	// Format the log message to be easy to read.
+	FString LogMessage = FString::Printf(TEXT("%s \n Code: %s \n Code String: %s \n Error: %s \n Stack: %s"), *Message, *ErrorCode, *ErrorCodeString, *ErrorMessage, *Stack);
+
+	ELogVerbosity::Type LogVerbosity = ELogVerbosity::Error;
+
+	// There are currently no logger names included in the errors from Spatial Service and so we always give it this LogCategory.
+	FString LogCategory = TEXT("SpatialService");
+
+	// Serialization must be done on the game thread.
+	AsyncTask(ENamedThreads::GameThread, [this, LogMessage, LogVerbosity, LogCategory]
+	{
+		Serialize(*LogMessage, LogVerbosity, FName(*LogCategory));
+	});
+}
+
 void SSpatialOutputLog::FormatAndPrintRawLogLine(const FString& LogLine)
 {
 	// Log lines have the format time=LOG_TIME level=LOG_LEVEL logger=LOG_CATEGORY msg=LOG_MESSAGE
-	const FRegexPattern LogPattern = FRegexPattern(TEXT("level=(.*) logger=(.*\\.)?(.*) msg=(.*)"));
+	const FRegexPattern LogPattern = FRegexPattern(TEXT("level=(.*) msg=(.*) loggerName=(.*\\.)?(.*)"));
 	FRegexMatcher LogMatcher(LogPattern, LogLine);
 
 	if (!LogMatcher.FindNext())
 	{
-		UE_LOG(LogSpatialOutputLog, Error, TEXT("Failed to parse log line: %s"), *LogLine);
+		// If this line could not be matched then it's most likely an error with a stack trace. Attempt to use the ErrorLine formatter.
```

Comment:
We can't be more specific about this?

Reply:
Tbh there is only two cases which is either a standard log line or a standard error line, I put "most likely" here because I can imagine a case one day where a user has worker logs with fields which breaks our regex and so it's not 100% but more like 99.99%..... I can just change this to:
`// If this log line did not match the log line regex then it is an error line which is parsed differently.`
Sound good?